contracts/polygon/connectors/gelato/aave-services/protection/main.sol

```diff
@@ -0,0 +1,155 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.7.6;
+pragma abicoder v2;
+
+/**
+ * @title Aave Protection.
+ * @dev Protect DSA against Liquidation risk on Aave with Gelato.
+ */
+
+
+import {IERC20} from "./interface.sol";
+import {Helpers} from "./helpers.sol";
+
+abstract contract GAaveProtectionResolver is Helpers {
+    /// @dev Function for submitting a protection task
+    /// @param _wantedHealthFactor targeted health after protection.
+    /// @param _minimumHealthFactor trigger protection when current health
+    /// factor is below _minimumHealthFactor.
+    /// @param _isPermanent boolean to set a protection as permanent
+    function submitProtection(
+        uint256 _wantedHealthFactor,
+        uint256 _minimumHealthFactor,
+        bool _isPermanent
+    ) external payable {
+        _submitProtection(
+            _wantedHealthFactor,
+            _minimumHealthFactor,
+            _isPermanent
```

Comment:
Can we add events for these 4 main functions?

Reply:
Yes i added 3 events ```LogSubmitProtection```, ```LogUpdateProtection``` and ```LogCancelProtection```, theses events will be emitted inside these 4 externals functions.

Reply:
Do you want another event for revoke allowance, maybe like ```event LogRevokeAllowance(address dsa, address action)```?